src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1717,21 +1717,21 @@ private static object FoldNeverOverflowBinaryOperators(BinaryOperatorKind kind,
                 case BinaryOperatorKind.FloatRemainder:
                     return valueLeft.SingleValue % valueRight.SingleValue;
                 case BinaryOperatorKind.IntLeftShift:
-                    return valueLeft.Int32Value << valueRight.Int32Value;
+                    return valueLeft.Int32Value << (valueRight.Int32Value % 32);
```

Comment:
I'm confused. Why is this needed given that the C# spec says in https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#shift-operators

> When the type of x is int or uint, the shift count is given by the low-order five bits of count. In other words, the shift count is computed from count & 0x1F.

Reply:
Excellent question.  It shouldn't be needed.  I was looking at the ECMA specification for the CLR, which makes it undefined behavior!  Since the C# compiler does not produce the required mask operation before shifting, we are producing incorrect code!  I'll investigate further.  We may need to change the CLR specification to have similar language for the shift instructions.

Reply:
ok, we do emit a mask operation.